Sources/EpoxyCore/SwiftUI/LayoutUtilities/IdealHeightMeasurementContainer.swift

```diff
@@ -0,0 +1,138 @@
+    // will be incorrect.
+    if bounds.size != latestMeasurementBoundsSize {
+      measureView()
+    }
+  }
+
+  // MARK: Private
+
+  private let context: SizingContext
+
+  /// The bounds size at the time of the latest measurement.
+  ///
+  /// Used to ensure we don't do extraneous measurements if the bounds haven't changed.
+  private var latestMeasurementBoundsSize: CGSize?
+
+  /// The most recently measured intrinsic content size of the `uiView`, else `nil` if it has not
+  /// yet been measured.
+  private var latestMeasuredSize: CGSize? = nil
+
+  private func updateView(from oldValue: UIViewType) {
+    guard uiView !== oldValue else { return }
+    oldValue.removeFromSuperview()
+    addSubview(uiView)
+    setUpConstraints()
+    setNeedsLayout()
+  }
+
+  private func setUpConstraints() {
+    uiView.translatesAutoresizingMaskIntoConstraints = false
+
+    let leading = uiView.leadingAnchor.constraint(equalTo: leadingAnchor)
+    let top = uiView.topAnchor.constraint(equalTo: topAnchor)
+
+    // These constraints won't be fulfilled when resizing, but it should be higher than any other
+    // layout priorities.
+    let constraintPriority = UILayoutPriority(rawValue: UILayoutPriority.required.rawValue - 1)
+
+    let trailing = uiView.trailingAnchor.constraint(equalTo: trailingAnchor)
+    trailing.priority = constraintPriority
+
+    let bottom = uiView.bottomAnchor.constraint(equalTo: bottomAnchor)
+    bottom.priority = constraintPriority
+
+    NSLayoutConstraint.activate([leading, top, trailing, bottom])
+  }
+
+  /// Measures the `uiView`, returning the resulting size and storing it in `latestMeasuredSize`.
+  @discardableResult
+  private func measureView() -> CGSize {
+    // On the first layout, use the `initialSize` to measure with a reasonable first attempt, as
+    // passing zero results in unusable sizes and also upsets SwiftUI.
+    let measurementBounds = bounds.size == .zero ? context.proposedSize : bounds.size
+    latestMeasurementBoundsSize = measurementBounds
+
+    let targetSize = CGSize(
+      width: measurementBounds.width,
+      height: UIViewType.layoutFittingCompressedSize.height)
+
+    let fittingSize = uiView.systemLayoutSizeFitting(
+      targetSize,
+      withHorizontalFittingPriority: .defaultHigh,
+      verticalFittingPriority: .fittingSizeLevel)
+
+    let measuredSize = CGSize(width: UIView.noIntrinsicMetric, height: fittingSize.height)
+
+    // We need to update the ideal size async otherwise we'll get the "Modifying state during view
+    // update, which will cause undefined behavior" runtime warning as the view's intrinsic content
+    // size is queried during the view update phase.
+    DispatchQueue.main.async { [idealHeight = context.idealHeight] in
+      idealHeight.wrappedValue = measuredSize.height
+    }
+
+    if latestMeasuredSize != measuredSize {
+      latestMeasuredSize = measuredSize
+      super.invalidateIntrinsicContentSize()
```

Comment:
calling super directly here so we don't keep measuring in an infinite loop

Reply:
```suggestion
      // We call super directly here so we don't keep measuring in an infinite loop
      super.invalidateIntrinsicContentSize()
```